frame/contracts/src/wasm/prepare.rs

```diff
@@ -1214,7 +1165,7 @@ mod tests {
 				(func (export "deploy"))
 			)
 			"#,
-			Err("use of floating point type in globals is forbidden")
```

Comment:
don't we want to keep the more specific error message?

Reply:
We can't. The function returns a `&'static str`. We opted for logging the error message to the console instead of returning it. Not optimal. The error handling there is a mess. But the whole thing will become much simpler once we have #13639. Then we can have a hard look on the error handling there.